packages/core/src/indexing-store/sqlite/store.ts

```diff
@@ -338,6 +342,196 @@ export class SqliteIndexingStore implements IndexingStore {
     });
   };
 
+  /*
+   * To determine correct sorting where rows =
+   * [
+   *    id: 0,
+   *    id: 1,
+   *    id: 2,
+   *    id: 3,
+   * ]
+   *
+   * If no after or before arguments, then sort as is;
+   *
+   * If after argument,
+   *   1) sort in same order as passed in
+   *   2) or if no orderBy argument sort as ascending;
+   *
+   * If before argument, then since our query needs to fetch in reverse
+   * (select * from x where id < before) we need to
+   *   1) make orderBy the opposite of the orderBy the user passed in and
+   *      then reverse() the results
+   *   2) or if no orderBy argument, sort in descending;
+   *
+   * after {id : 1} : (select * from x where id > before order by 'asc')
+   * [
+   *    id: 2,
+   *    id: 3,
+   * ]
+   *
+   * before { id: 2} : (select * from x where id < before order by 'desc')
+   * 1)
+   * [
+   *    id: 1,
+   *    id: 0
+   * ]
+   * 2) reverse()
+   * [
+   *    id: 0,
+   *    id: 1
+   * ]
+   */
+
+  findManyPaginated = async ({
+    tableName,
+    checkpoint = "latest",
+    where,
+    before,
+    after,
+    take,
+    orderBy,
+  }: {
+    tableName: string;
+    checkpoint?: Checkpoint | "latest";
+    where?: WhereInput<any>;
+    before?: string;
+    after?: string;
+    take?: number;
+    orderBy?: OrderByInput<any>;
+  }) => {
+    return this.wrap({ method: "findMany", tableName }, async () => {
+      const table = `${tableName}_versioned`;
+
+      let query = this.db.selectFrom(table).selectAll();
+
+      if (checkpoint === "latest") {
+        query = query.where("effectiveToCheckpoint", "=", "latest");
+      } else {
+        const encodedCheckpoint = encodeCheckpoint(checkpoint);
+        query = query
+          .where("effectiveFromCheckpoint", "<=", encodedCheckpoint)
+          .where(({ eb, or }) =>
+            or([
+              eb("effectiveToCheckpoint", ">", encodedCheckpoint),
+              eb("effectiveToCheckpoint", "=", "latest"),
+            ]),
+          );
+      }
+
+      if (where) {
+        const whereConditions = buildSqlWhereConditions({
+          where,
+          encodeBigInts: true,
+        });
+        for (const whereCondition of whereConditions) {
+          query = query.where(...whereCondition);
+        }
+      }
+
+      if (take) {
+        const limit = validateTake(take);
+        // Get +1 extra row to determine if has next
+        query = query.limit(limit + 1);
+      }
+
+      let orderDirection = "asc";
+
+      if (orderBy) {
+        // Get reversed order conditions, for finding the -1 row
+        const orderByConditions = !before
+          ? buildSqlOrderByConditions({ orderBy })
+          : buildSqlOrderByConditionsReversed({ orderBy });
```

Comment:
Could we use the existing `buildSqlOrderByConditions` and then reverse the direction(s) here?